programs/vote/src/vote_state/mod.rs

```diff
@@ -1885,12 +1919,535 @@ mod tests {
+                    (48, 14),
+                    (49, 13),
+                    (50, 12),
+                    (51, 11),
+                    (52, 10),
+                    (53, 9),
+                    (54, 8),
+                    (55, 7),
+                    (56, 6),
+                    (57, 5),
+                    (58, 4),
+                    (59, 3),
+                    (60, 2),
+                    (61, 1),
+                ],
+                Some(30),
+                30,
+            ),
+            // 31 slots finalized
+            (
+                vec![
+                    (1, 31),
+                    (2, 30),
+                    (3, 29),
+                    (4, 28),
+                    (5, 27),
+                    (6, 26),
+                    (7, 25),
+                    (8, 24),
+                    (9, 23),
+                    (10, 22),
+                    (11, 21),
+                    (12, 20),
+                    (13, 19),
+                    (14, 18),
+                    (15, 17),
+                    (16, 16),
+                    (17, 15),
+                    (18, 14),
+                    (19, 13),
+                    (20, 12),
+                    (21, 11),
+                    (22, 10),
+                    (23, 9),
+                    (24, 8),
+                    (25, 7),
+                    (26, 6),
+                    (27, 5),
+                    (28, 4),
+                    (29, 3),
+                    (30, 2),
+                    (31, 1),
+                ],
+                Some(0),
+                vec![
+                    (32, 31),
+                    (33, 30),
+                    (34, 29),
+                    (35, 28),
+                    (36, 27),
+                    (37, 26),
+                    (38, 25),
+                    (39, 24),
+                    (40, 23),
+                    (41, 22),
+                    (42, 21),
+                    (43, 20),
+                    (44, 19),
+                    (45, 18),
+                    (46, 17),
+                    (47, 16),
+                    (48, 15),
+                    (49, 14),
+                    (50, 13),
+                    (51, 12),
+                    (52, 11),
+                    (53, 10),
+                    (54, 9),
+                    (55, 8),
+                    (56, 7),
+                    (57, 6),
+                    (58, 5),
+                    (59, 4),
+                    (60, 3),
+                    (61, 2),
+                    (62, 1),
+                ],
+                Some(31),
+                31,
+            ),
+            // Complex expiry: start with 1 - 31 in the tower, then locally vote for 35, 36, 37, 38, 42, 43, 46, 47,
+            // 48, 49, 50, 53.  None of the intermediate tower states land as tx, only the resulting complete state
+            // does.  This test very succinctly illustrates the way that "timely vote credits" are an approximation of
```

Comment:
Much of this comment is out of date, it was relevant when this change included the "timely vote credits" stuff.  I can remove it if you like.  I *hope* that it's just prep work for a future change though ....

Reply:
let's only keep the relevant comments for now